src/vcpkg/build.cpp

```diff
@@ -207,25 +238,35 @@ namespace vcpkg::Build
 
         Checks::check_exit(VCPKG_LINE_INFO, action != nullptr);
         ASSUME(action != nullptr);
+        auto& scf = *action->source_control_file_and_location.value_or_exit(VCPKG_LINE_INFO).source_control_file;
+        const auto& spec_name = spec.name();
+        const auto& core_paragraph_name = scf.core_paragraph->name;
+        if (spec_name != core_paragraph_name)
+        {
+            Checks::msg_exit_with_error(VCPKG_LINE_INFO,
+                                        msgSourceFieldPortNameMismatch,
+                                        msg::package_name = core_paragraph_name,
+                                        msg::path = spec_name);
+        }
+
         action->build_options = default_build_package_options;
         action->build_options.editable = Editable::YES;
         action->build_options.clean_buildtrees = CleanBuildtrees::NO;
         action->build_options.clean_packages = CleanPackages::NO;
 
         const auto build_timer = ElapsedTimer::create_started();
         const auto result = Build::build_package(args, paths, *action, binary_cache, build_logs_recorder, status_db);
-        print2("Elapsed time for package ", spec, ": ", build_timer, '\n');
-
+        msg::print(msgElapsedForPackage, msg::spec = spec, msg::elapsed = build_timer.elapsed());
         if (result.code == BuildResult::CASCADED_DUE_TO_MISSING_DEPENDENCIES)
         {
-            print2(Color::error, "The build command requires all dependencies to be already installed.\n");
-            print2("The following dependencies are missing:\n\n");
+            LocalizedString errorMsg =
+                msg::format(msg::msgErrorMessage).append(msg::format(msgBuildDependenciesMissing));
```

Comment:
```suggestion
                msg::format(msg::msgErrorMessage).append(msgBuildDependenciesMissing);
```

append has formatting built in